src/extend.rs

```diff
@@ -1,11 +1,25 @@
+//! Structured extensions for building and parsing Thing Descriptions
+//!
+//! The Thing Description can be extended adding additional ontologies in its [@context](https://www.w3.org/TR/json-ld11/#the-context).
+//!
+//! This module provides a trait, [ExtendableThing], to define extensions for each of the standard
+//! elements of a description.
+
 use serde::{Deserialize, Serialize};
 
 use crate::hlist::{Cons, Nil};
 
+/// Requirement trait for extending a Thing Description alement
+///
+/// The structure must be serializable/deserializable use `#[serde(skip)]` if it has to hold data
+/// you do not want to push in the serialized description.
 pub trait ExtendablePiece: Serialize + for<'a> Deserialize<'a> {}
 
 impl<T> ExtendablePiece for T where T: Serialize + for<'a> Deserialize<'a> {}
 
+/// Main extension trait
+///
+/// The trait uses an associated type for each element of the ThingDescription
```

Comment:
Maybe we should expand this. It is not trivial to understand that these associated types will be _added_ to `other` fields for all the nested levels of the thing.

Reply:
We will, this is just the first pass of many I'm afraid :)